src/amqp.js

```diff
@@ -0,0 +1,340 @@
+        {
+          correlationId: blobId,
+          persistent: true
+        }
+      );
+      debugDev(`Send message to queue: ${queue}`);
+
+      return true;
+    } catch (error) {
+      const errorToThrow = error;
+      //if (error instanceof ApiError) {
+      //  throw error;
+      //}
+      debug(`SendToQueue errored: ${JSON.stringify(error)}`);
+      handleAmqpErrors(errorToThrow);
+    }
+  }
+
+  async function removeQueue({blobId, status}) {
+    const queue = generateQueueId({blobId, status});
+    // deleteQueue borks the channel if the queue does not exist
+    // -> use throwaway tempChannel to avoid killing actual channel in use
+    // this might be doable also with assertQueue before deleteQueue
+    const tempChannel = await connection.createChannel();
+    debug(`Removing queue ${queue}.`);
+    await tempChannel.deleteQueue(queue);
+
+    if (tempChannel) {
+      await tempChannel.close();
+      return;
+    }
+
+    return;
+  }
+
+  // ----------------
+  // Helper functions
+  // ----------------
+
+  function checkMessageCount(channelInfo) {
+    if (channelInfo.messageCount < 1) {
+      debug(`checkQueue: ${channelInfo.messageCount} - queue is empty`);
+      return false;
+    }
+    debug(`Queue has ${channelInfo.messageCount} messages`);
+    return true;
+  }
+
+  function generateQueueId({blobId, status}) {
+    if (blobId === undefined || status === undefined || blobId.length < 1 || status.length < 1) {
+      throw new ApiError(httpStatus.INTERNAL_SERVER_ERROR, 'invalid operation parameters');
+    }
+
+    return `${blobId}.${status}`;
+  }
+
+  function messagesToRecords(messages) {
+    debug(`Parsing messages (${messages.length}) to records`);
+
+    return messages.map(message => {
+      const content = JSON.parse(message.content.toString());
+      // Use subfieldValues: false validationOption here
+      return new MarcRecord(content.data, {subfieldValues: false});
+    });
+  }
+
+  async function getData(queue) {
+    debug(`Getting queue data from ${queue}`);
+    try {
+      const {messageCount} = await channel.checkQueue(queue);
+      debug(`There is ${messageCount} messages in queue ${queue}`);
+      const messagesToGet = messageCount >= CHUNK_SIZE ? CHUNK_SIZE : messageCount;
+      debug(`Getting ${messagesToGet} messages from queue ${queue}`);
+
+      const messages = await pump(messagesToGet);
+
+      debug(`Returning ${messages.length} unique messages`);
+
+      return messages;
+    } catch (error) {
+      handleAmqpErrors(error);
+    }
+
+    async function pump(count, results = [], identifiers = []) {
+      if (count === 0) {
+        return results;
+      }
+
+      const message = await channel.get(queue);
+      const identifier = {
+        correlationId: message.properties.correlationId,
+        deliveryTag: message.fields.deliveryTag
+      };
+      // Filter not unique messages
+      if (identifiers.includes(identifier)) {
```

Comment:
Object level comparison does not work with Array.includes based on quick inspection in local node runtime:
```
> [{'a': 1}].includes({'a': 1})
false
```